packages/core/src/columns/columnType.ts

```diff
@@ -594,6 +634,38 @@ export abstract class ColumnTypeBase<
     }) as unknown as ParseColumn<T, Output>;
   }
 
+  /**
+   * Mark the column as to have specific Typescript type.
+   *
+   * ```ts
+   * export class Table extends BaseTable {
+   *   readonly table = 'table';
+   *   columns = this.setColumns((t) => ({
+   *     column: t.text().asType(columnType<"foo" | "bar">()),
+   *   }));
+   * }
+   *
+   * // value will be typed as "foo" | "bar"
+   * const value = await db.table.get('column');
+   *
+   * To alter base, input, output and query types individually, pass them as generic parameters:
+   *
+   * ```ts
+   * columnType<Type, InputType = Type, OutputType = Type, QueryType = Type>()
```

Comment:
The doc should explain what these types are for. I usually first write the docs and then copy-paste them to the methods.

- The first `Type` is the base one, used as a default for other types.
- `InputType` is for `create`, `update` methods.
- `OutputType` is for the data that is loaded from a database and parsed if the column has `parse`. -
- `QueryType` is used in `where` and other query methods, it should be compatible with the actual database column type.

`InputType = Type` isn't a valid TS, so I suggest just listing them with a comma.

```suggestion
   * columnType<Type, InputType, OutputType, QueryType>()
```